numpy/lib/function_base.py

```diff
@@ -115,13 +124,28 @@ def scott(x):
         Scott Estimator
         The binwidth is proportional to the standard deviation of the data and
         inversely proportional to the cube root of data size (asymptotically optimal)
-
         """
         h = 3.5 * x.std() * x.size ** (-1.0 / 3)
         if h > 0:
             return np.ceil(x.ptp() / h)
         return 1
 
+    def doane(x):
+        """
+        Doane's Estimator
+        Improved version of Sturges' formula which works better for non-normal
+        data.
+        See http://stats.stackexchange.com/questions/55134/doanes-formula-for-histogram-binning
+        """
+        if x.size > 2:
+            stdskew = np.sqrt(6.0 * (x.size - 2) / ((x.size + 1.0) * (x.size + 3)))
+            sigma = np.std(x)
+            if sigma > 0:
+                mu = np.mean(x)
+                skew = (np.mean(x**3) - 3 * mu * sigma**2 - mu**3) / sigma**3
```

Comment:
I am not a specialist. Honestly, I wouldn't mind another reference, but I guess I trust you (and that stackexchange guy) mostly, though the fact that it seems hard to almost find the right formula makes me wonder a bit.

I checked the R moments package, they use the "simple" formula: `(sum((x - mean(x))^3)/n)/(sum((x - mean(x))^2)/n)^(3/2)` for the skewness. My guess is still that it is numerically slightly better because x and x.mean() should be on a similar magnitude, so that the subtractions can't have catastrophic loss of precision, but maybe I am thinking about it the wrong way, it is just a guess.

@josef-pkt do you know of the top of your head that this formula is the right "Doane's estimator"?